topics/icrc_28_trusted_origins.md

```diff
@@ -0,0 +1,161 @@
+# ICRC-28: Trusted Origins
+
+[![Status Badge](https://img.shields.io/badge/STATUS-DRAFT-ffcc00.svg)](https://github.com/orgs/dfinity/projects/31)
+[![Standard Issue](https://img.shields.io/badge/ISSUE-ICRC--28-blue?logo=github)](https://github.com/dfinity/wg-identity-authentication/issues/115)
+
+<!-- TOC -->
+* [ICRC-28: Trusted Origins](#icrc-28-trusted-origins)
+  * [Summary](#summary)
+  * [Origin](#origin)
+  * [Get trusted origins](#get-trusted-origins)
+    * [Canister metadata](#canister-metadata)
+    * [Canister method](#canister-method)
+  * [Use-Cases](#use-cases)
+    * [Hot Signer Use-Case](#hot-signer-use-case)
+    * [Cold Signer Use-Case](#cold-signer-use-case)
+<!-- TOC -->
+
+## Summary
+
+Relying parties wants to request delegations from signers that can make authenticated calls to their own canisters,
+while at the same time still requiring user approval to make authenticated calls to canisters it doesn't own.
+
+**Example**
+
+- DSCVR requests a delegation that can make authenticated calls to its own canisters: dscvr_can_1 and dscvr_can_2
+- When DSCVR wants the user to transfer tokens, it will need to request the user's approval in a signer prompt
+
+For a signer to verify if a relying party should receive a delegation for a given list of canister targets,
+it will need to check for each canister target if the relying party is trusted.
+
+## Origin
+
+The window origin (as defined in https://developer.mozilla.org/en-US/docs/Glossary/Origin) is used to identify one
+relying party from another.
+
+> The origin is the concatenation of the protocol and "://", the host name if one exists, and ":" followed by a port
+> number if a port is present and differs from the default port for the given protocol. Examples of typical origins
+> are https://example.org (implying port 443), http://example.net (implying port 80), and http://example.com:8080.
+
+The origin can't be trusted with some transport layers. It's up to the signer to implement a trustable way to identify
+the origin of the relying party.
+
+The origin **MUST** be identified in a trustable way by the signer for example through `postMessage`.
+
+## Get trusted origins
+
+A canister **MUST** make both a metadata entry and a canister method available, both **MUST** return the same list of
+trusted origins.
+
+### Canister metadata
+
+Allows for tooling to pull the list of trusted origins out of the state tree from metadata
+section `icp:public icrc28_trusted_origins`.
+
+This is the recommended approach for signers since it does not require a canister call to each delegation target
+canister.
+Instead, it can get the trusted origins from the metadata section and verify the certificate.
+
+### Canister method
+
+Returns list of trusted origins.
+
+This an alternative approach for canisters that want to get the list of trusted origins.
+
+```
+icrc28_get_trusted_origins : () -> (vec text);
```

Comment:
Do we also need an `icrc28_supported_standards` hook for the inevitable extensions?

At this point I'm really wondering whether we should not spearhead an effort to simply call it `icrc_supported_standards` and unify that across all the variants... 🤔

Reply:
> imply call it icrc_supported_standards and unify that across all the variants..
I could imagine the method might be different at some point for an ICRC standard. Maybe a `icrc_supported_base_standards` that only lists base standards, not extensions. This could be it's own ICRC standard though e.g. `icrc58_supported_base_standards`.

Reply:
Added supported standards.